actions/websocket_actions.jsx

```diff
@@ -350,6 +355,60 @@ function handleUpdateTeamEvent(msg) {
     TeamStore.updateTeam(msg.data.team);
 }
 
+function handleDeleteTeamEvent(msg) {
+    const deletedTeam = JSON.parse(msg.data.team);
+    const state = store.getState();
+    const {teams} = state.entities.teams;
+    if (
+        deletedTeam &&
+        teams &&
+        teams[deletedTeam.id] &&
+        teams[deletedTeam.id].delete_at === 0
+    ) {
+        const {currentUserId} = state.entities.users;
+        const {currentTeamId, myMembers} = state.entities.teams;
+        const teamMembers = Object.values(myMembers);
+        const teamMember = teamMembers.find((m) => m.user_id === currentUserId && m.team_id === currentTeamId);
+
+        let newTeamId = '';
+        if (
+            deletedTeam &&
+            teamMember &&
+            deletedTeam.id === teamMember.team_id
+        ) {
+            const myTeams = {};
+            getMyTeams(state).forEach((t) => {
+                myTeams[t.id] = t;
+            });
+
+            for (let i = 0; i < teamMembers.length; i++) {
+                const memberTeamId = teamMembers[i].team_id;
+                if (
+                    myTeams &&
+                    myTeams[memberTeamId] &&
+                    myTeams[memberTeamId].delete_at === 0 &&
+                    deletedTeam.id !== memberTeamId
+                ) {
+                    newTeamId = memberTeamId;
+                    break;
+                }
+            }
+        }
```

Comment:
This is exactly the type of thing reducers are meant to do, any time a team is deleted you're going to want this to happen, right? Otherwise any time you dispatch a team deletion event you need to manually do all this work again

Reply:
I'll see what I can do when I work with the mobile ticket - [ICU-685](https://mattermost.atlassian.net/browse/ICU-685)

Reply:
I should say this is critical for the release, since it breaks the webapp whenever there is deleted team.

Reply:
Ok, I'll approve then